src/output.cpp

```diff
@@ -586,11 +604,18 @@ void JsonOutput::map_stats(BPFtrace &bpftrace, IMap &map,
     out_ << "{";
 
   uint32_t i = 0;
+  uint32_t j = 0;
   for (auto &key_count : total_counts_by_key)
   {
     auto &key = key_count.first;
     auto &value = values_by_key.at(key);
 
+    if (map.type_.IsAvgTy() && top)
```

Comment:
Any reason we restrict to `IsAvgTy` here? Seems stats maps is still broken. Maybe reword the PR so both `avg` and `stats` is fixed.

Reply:
I suppose `top` might be ambiguous for `stats`. `stats` returns the average, the sum and the count but `top` does not specify which statistic it's based on. I noticed that `print_map_stats` would sort its entries based on the average but the user might not necessarily expect that. As for `div`, it would somehow be strange, too: if use `div` on all three statistics, then the output does not have avg * count == sum anymore; also it's likely that count become 0 after dividing div. I'm thinking if we should disallow top&div for stats maps during semantic checking.

Reply:
And I wonder if there is an easy way to test it. I'm thinking about have the following to work as the testcase:
```
bpftrace -e 'i:ms:1 {@[nsecs / 1000000 % 10] = avg(1);} i:s:1 { print(@, 3, 0); clear(@); exit();}'
```
Not sure if I could come up with a regular expression to verify the output.

I notice there doesn't seem to be runtime tests for `print(map, top, div)` (or I missed them out); there could be another pr for runtime tests if there aren't. Any thoughts?